File: backend/src/main/java/com/awais/hr/module/engagement/service/EngagementService.java
package com.awais.hr.module.engagement.service;

import java.util.List;
import java.util.Map;

public interface EngagementService {
    List<Map<String, Object>> getSurveys();
    Map<String, Object> createSurvey(Map<String, Object> body);
    List<Map<String, Object>> getRecognitions();
    Map<String, Object> sendRecognition(String senderEmail, Map<String, Object> body);
    List<Map<String, Object>> getSuggestions();
    Map<String, Object> submitSuggestion(Map<String, Object> body);
}

File: frontend/src/services/engagementService.js
import apiClient from './api';

export const getSurveys = () => apiClient.get('/suite/engagement/surveys');
export const createSurvey = (dto) => apiClient.post('/suite/engagement/surveys', dto);
export const getRecognitions = () => apiClient.get('/suite/engagement/recognitions');
export const sendRecognition = (dto) => apiClient.post('/suite/engagement/recognitions', dto);
export const getSuggestions = () => apiClient.get('/suite/engagement/suggestions');
export const submitSuggestion = (dto) => apiClient.post('/suite/engagement/suggestions', dto);
